src/zenml/zen_server/cloud_utils.py

```diff
@@ -196,29 +199,49 @@ def _fetch_auth_token(self) -> str:
         Returns:
             Auth token.
         """
+        if (
+            self._token is not None
+            and self._token_expires_at is not None
+            and datetime.now(timezone.utc) + timedelta(minutes=5)
+            < self._token_expires_at
+        ):
+            return self._token
+
         # Get an auth token from auth0
-        auth0_url = f"https://{self._config.auth0_domain}/oauth/token"
+        login_url = f"{self._config.api_url}/auth/login"
```

Comment:
Maybe we need to keep the full URL configurable, so one can set it to an older value as well?
I mean now we are locking ourselves into in-house implementation, which is not a bad thing per se.  So you can also ignore this.

Reply:
I agree, but I'm trying to keep the current tenant environment variables still valid for now, and the API URL we already have in its base form.

To support fully configurable URLs, I would have to make changes to the tenant deployment configurations while also supporting backwards-compatibility for older tenants, which makes this more difficult.